core/src/main/scala/akka/kafka/internal/KafkaConsumerActor.scala

```diff
@@ -506,27 +509,21 @@ import scala.util.control.NonFatal
                                 exception: Exception): Unit = {
           // this is invoked on the thread calling consumer.poll which will always be the actor, so it is safe
           val duration = System.nanoTime() - startTime
+          if (exception == null) {
+            self ! Committed(offsets.asScala.toMap)
+          } else {
+            log.error("Kafka commit failed", exception)
```

Comment:
The current Alpakka Kafka behaviour is to return a failure as the reply to the committer.  

```
if (exception != null) sendReply(Status.Failure(exception))
```

IIUC it is up to the user to handle the `Failure` of the `Future` that was returned by the commit.  In `Committer.batchFlow` it would just return the `Failure` and kill the stream.  We can change this to fail the stream to remain consistent.

WRT to retrying commits:

In cases where we can retry sending the commit we'll need to be careful we don't end up committing out of order.  I recall discussing this with someone (maybe the Alpakka team?)  It's discussed in Kafka: The Definitive Guide and summarized in this SO post:

https://stackoverflow.com/questions/53240589/kafka-commitasync-retries-with-commit-order